qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -556,3 +558,61 @@ def test_metadata():
     with pytest.raises(ValueError, match=match):
         for tag, value in sorry_metadata.items():
             ds1.add_metadata(tag, value)
+
+
+class TestGetData:
+    x = ParamSpec("x", paramtype='numeric')
+    n_vals = 5
+    xvals = list(range(n_vals))
```

Comment:
The list should be redundant when you only use this to iterate over?

Reply:
yes, you are right. But I am also trying to convey a message that if for whatever reason somebody wants to use `xvals = [345, 456, 22, 11, 4]`, they can (as long as they keep the value of `n_vals` in sync) :)

Reply:
Ok